### PR TITLE
v4/v5: Add responsive containers

### PR DESCRIPTION
**Pull request by mdo**
Pulling this from @browner12's PR to simplify everything and document the new classes.

### CODE REVIEW

**Comment by XhmikosR** [comment]
@mdo this shouldn't have been merged without review...

**Comment by MartijnCuppens** [comment]
This is also a new feature and shouldn't be backported to `v4` either.

File: scss/_navbar.scss [hunk 1]
```diff
@@ -140,8 +139,7 @@
 
     &#{$infix} {
       @include media-breakpoint-down($breakpoint) {
-        > .container,
-        > .container-fluid {
+        > [class^=".container"] {
```
Comment (voltaek) [inline scss/_navbar.scss]: This shouldn't have a period in the class name text it's looking for.

File: scss/_navbar.scss [hunk 2]
```diff
@@ -165,8 +163,7 @@
         }
 
         // For nesting containers, have to redeclare for alignment purposes
-        > .container,
-        > .container-fluid {
+        > [class^=".container"] {
```
Comment (voltaek) [inline scss/_navbar.scss]: This also shouldn't have a period in the class name text it's looking for.

**Review by voltaek — COMMENTED** [review]
A couple typos in `navbar.scss`

**Comment by XhmikosR** [comment]
Also this shouldn't have been squashed like this.

**Comment by browner12** [comment]
I hate to say this, because I appreciate all the hard work that everyone puts into this project, but I am very disappointed with where this PR went.

I presented a PR a year and a half ago that solved an incredibly common problem, one that I've run into in almost every project I've ever done, and it's now turned into something that I, and I believe most other people, will have zero to no use for.

Containers are meant to be responsive or fluid. My PR allowed you to choose at which breakpoint this happens.  I laid out an **incredibly** common use case. It received tremendous support.

These changes now make the containers responsive, fluid, or fixed. `.container` will behave differently than `.container-md` which will lead to confusion.

I mean no disrespect to any of you, but I believe this needs to be readdressed.

**Comment by mdo** [comment]
@browner12 I legit screwed up bringing making changes to your PR and oversimplified the code here. I did want your original changes, and have stubbed out another PR to fix the responsive containers. See #29118.

**Pull request merged**